### PR TITLE
Docker container init check requirements

### PR DESCRIPTION
**Pull request by pilou-**
##### SUMMARY
`docker_container` module:
- check version of `docker-py` dependency too
- fix compatibility with `docker-py<2.2`
  Using this playbook with docker-py<2.2:
  ```
  - hosts: localhost
    gather_facts: no
    tasks:
      - name: Create a container
        docker_container:
          name: testcontainer
          hostname: testcontainer
          image: debian:stretch
          init: yes
  ```
  the following exception occurs:

  ```
  Traceback (most recent call last):",
    File /tmp/ansible_n0psgms7/ansible_module_docker_container.py, line 2081, in <module>",
      main()",
    File /tmp/ansible_n0psgms7/ansible_module_docker_container.py, line 2076, in main",
      cm = ContainerManager(client)",
    File /tmp/ansible_n0psgms7/ansible_module_docker_container.py, line 1703, in __init__",
      self.present(state)",
    File /tmp/ansible_n0psgms7/ansible_module_docker_container.py, line 1723, in present",
      new_container = self.container_create(self.parameters.image, self.parameters.create_parameters)",
    File /tmp/ansible_n0psgms7/ansible_module_docker_container.py, line 825, in create_parameters",
      host_config=self._host_config(),",
    File /tmp/ansible_n0psgms7/ansible_module_docker_container.py, line 931, in _host_config",
      return self.client.create_host_config(**params)",
    File lib/python3.6/site-packages/docker/api/container.py, line 596, in create_host_config",
      return HostConfig(*args, **kwargs)",
  TypeError: __init__() got an unexpected keyword argument 'init'"
  ```

Related: #34547.

This pull-request should be backported on 2.6.

##### ISSUE TYPE
 - Bugfix Pull Request

##### COMPONENT NAME
docker_container

##### ANSIBLE VERSION
```
ansible 2.7.0dev0 (devel 726d7fe4c3) last updated 2018/05/29 10:06:10 (GMT +200)
```

### CODE REVIEW

**Comment by pilou-** [comment]
Rebased in order to trigger a new shippable run.

**Comment by pilou-** [comment]
Rebased in order to get rid of `stale_ci` label.

**Review by resmo — APPROVED** [review]
Hit this issue, applied patch fixes it.

**Review by kassiansun — APPROVED** [review]
shipit

**Comment by resmo** [comment]
shipit

File: lib/ansible/module_utils/docker_common.py
Comment (resmo) [inline lib/ansible/module_utils/docker_common.py:192]: proabably one note, since this is "common" code for docker, I have concerns that there are other things depending on this, we should probably keep this code.

Comment (kassiansun) [reply lib/ansible/module_utils/docker_common.py:192]: `init` and many other options are for `docker_container` only, we should not have put version detection here.

**Comment by felixfontein** [comment]
It would be nice if this could go into 2.6.x ASAP, as 2.6.0 currently doesn't work for some people and this probably fixes it (on first glance); see #42162.

File: lib/ansible/module_utils/docker_common.py
Comment (felixfontein) [reply lib/ansible/module_utils/docker_common.py:192]: This version check was added for `docker_container`'s `init` (in #34547) only recently (May 24).

**Comment by resmo** [comment]
let me ping @maxamillion to look over it.

File: lib/ansible/modules/cloud/docker/docker_container.py
Comment (felixfontein) [inline lib/ansible/modules/cloud/docker/docker_container.py]: Shouldn't this be a `>` test?

File: lib/ansible/modules/cloud/docker/docker_container.py
Comment (felixfontein) [inline lib/ansible/modules/cloud/docker/docker_container.py]: Same here.

**Review by felixfontein — CHANGES_REQUESTED** [review]

File: lib/ansible/modules/cloud/docker/docker_container.py
Comment (felixfontein) [reply lib/ansible/modules/cloud/docker/docker_container.py]: (Actually, `>=`.)

**Comment by kassiansun** [comment]
#42274 is blocked by this currently, @pilou- could you check the review comments?

File: lib/ansible/modules/cloud/docker/docker_container.py
Comment (dmke) [reply lib/ansible/modules/cloud/docker/docker_container.py]: Can confirm, the tests should be `>=`. With those modifications, I have sucessfully applied this patch to v2.6.0 (8ce3bd6dea).

**Comment by pilou-** [comment]
@felixfontein fixed, thanks for the review !

Note there was another typo (`s/self.params/self.module.params/`) :-/

Manual tests (using Docker API version 1.37):
* using `init` parameter:
    * docker-py 1.10.6 (failure is expected)
        ```
        TASK [Create a container] ***
        fatal: [localhost]: FAILED! => {"changed": false, "msg": "docker-py version is 1.10.6. Minimum version required is 2.2 to set init option."}
        ```
    * docker 2.1.0 (failure is expected)
        ```
        TASK [Create a container] ***
        fatal: [localhost]: FAILED! => {"changed": false, "msg": "docker-py version is 2.1.0. Minimum version required is 2.2 to set init option."}
        ```
    * docker 3.4.1
        ```
        TASK [Create a container] ***
        changed: [localhost]
        ```
* without `init` parameter: no error with the 3 versions

**Review by felixfontein — APPROVED** [review]
LGTM (haven't tested it, though)

**Comment by resmo** [comment]
glad we got these issues fixed before merge. ;)

**Comment by resmo** [comment]
shipit

**Comment by mustafayildirim** [comment]
Is there any progress?

**Comment by felixfontein** [comment]
bot_status

**Comment by kassiansun** [comment]
shipit

**Comment by felixfontein** [comment]
@resmo: will you create a backport PR for 2.6?

**Comment by resmo** [comment]
@felixfontein yes of course :) I will create it later today.

**Pull request merged**
